### PR TITLE
 macOS: Implement texture converter for Metal 

### PR DESCRIPTION
**Pull request by nastys**
Implements texture converters for macOS for the following formats:
- CELL_GCM_TEXTURE_R5G6B5
- CELL_GCM_TEXTURE_R6G5B5
- CELL_GCM_TEXTURE_D1R5G5B5
- CELL_GCM_TEXTURE_A1R5G5B5
- CELL_GCM_TEXTURE_A4R4G4B4
- CELL_GCM_TEXTURE_R5G5B5A1

### CODE REVIEW

File: rpcs3/Emu/RSX/Common/TextureUtils.cpp
Comment (elad335) [inline rpcs3/Emu/RSX/Common/TextureUtils.cpp:37]: Uh, I am convinced that this needs at least be vectorized otherwise it'll be sluggish.

Comment (Nekotekina) [reply rpcs3/Emu/RSX/Common/TextureUtils.cpp:37]: Don't bother, it should be really trivial for auto-vectorization in clang/gcc (or at least with minimal changes)

**Review by kd-11 — APPROVED** [review]
There's still a bunch of stuff missing, specifically w.r.t image readback and GPU-side format conversions (WCB, Blit engine), but that requires some design changes to work properly, and also fortunately rarely used with 16-bit formats.
Needs a rebase before merge, lgtm otherwise.

**Pull request closed without merging**
